flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java

```diff
@@ -455,6 +455,15 @@ protected void restartTasks(final Set<ExecutionVertexID> verticesToRestart) {
 			this.delegate = checkNotNull(delegate);
 		}
 
+		@Override
+		public CompletableFuture<LogicalSlot> allocateBatchSlot(
```

Comment:
Is this necessary?

Reply:
Yes. Without this method override, the default implementation of the super class will be called, which is based on SlotProvider#allocateBatchSlot. The default implementation throws an UnsupportedOperationException, which will cause some UT to fail.

Reply:
Thanks for the clarification!